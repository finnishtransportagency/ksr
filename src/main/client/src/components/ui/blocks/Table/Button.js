import styled, { css } from 'styled-components';
import * as styles from '../../defaultStyles';

const Button = styled.div`
    background: ${styles.colorBackgroundDark};
    display: flex;
    height: 60px;
    align-items: center;
    justify-content: center;
    font-size: 1.25em;
    left: 0;
    visibility: hidden;
    width: 60px;
    
    &:hover {
        cursor: pointer;
        background: ${styles.colorBackgroundDarkSecondary};
    };
    
    ${props => props.tableOpen && css`
        left: 0;
        position: relative;
        visibility: visible;
    `};
    
    ${props => props.toggleButton && css`
        visibility: visible;
    `};
    
    ${props => props.disabled && css`
        color: ${styles.colorFontDisabled};
        
        &:hover {
            cursor: default;
            background: ${styles.colorBackgroundDark};
        };
    `};
`;

export default Button;
