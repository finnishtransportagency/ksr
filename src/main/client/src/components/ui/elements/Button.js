import styled, { css } from 'styled-components';
import * as styles from '../defaultStyles';

export const Button = styled.button`
    font-weight: 500;
    font-size: 16px;
    border: 0;
    padding: 1rem;
    transition: 0.3s;
    margin: 1em 0;
    color: ${styles.colorFontLight};
    background: ${styles.colorMain};
    -webkit-box-shadow: ${styles.shadowDefault};
    -moz-box-shadow: ${styles.shadowDefault};
    box-shadow: ${styles.shadowDefault};
    
    ${props => props.flat && css`
         color: ${styles.colorMainLight};
         background: none;
         -webkit-box-shadow: none;
         -moz-box-shadow: none;
         box-shadow: none;
    `}
    
    ${props => props.disabled && css`
         opacity: 0.5;
    `}
    
    ${props => props.wide && css`
        width: 100%;
    `}
    
    &:hover {
        @media screen and (-ms-high-contrast: active), (-ms-high-contrast: none) {
            /* IE11 Fix, because it does not support media hover */
            cursor: pointer;
            -webkit-box-shadow: ${styles.shadowDark};
            -moz-box-shadow: ${styles.shadowDark};
            box-shadow: ${styles.shadowDark};
            background: ${styles.colorMainDark};
            
            ${props => props.disabled && css`
                -webkit-box-shadow: ${styles.shadowDefault};
                -moz-box-shadow: ${styles.shadowDefault};
                box-shadow: ${styles.shadowDefault};
                opacity: 0.5;
                background: ${styles.colorMain};
                cursor: not-allowed;
            `}
            
            ${props => props.flat && css`
                -webkit-box-shadow: none;
                -moz-box-shadow: none;
                box-shadow: none;
                background: rgba(0, 0, 0, 0.1);
            `}
            
            ${props => props.flat && props.disabled && css`
                background: none;
            `}
        }
        @media (hover) {
            cursor: pointer;
            -webkit-box-shadow: ${styles.shadowDark};
            -moz-box-shadow: ${styles.shadowDark};
            box-shadow: ${styles.shadowDark};
            background: ${styles.colorMainDark};
            
            ${props => props.disabled && css`
                -webkit-box-shadow: ${styles.shadowDefault};
                -moz-box-shadow: ${styles.shadowDefault};
                box-shadow: ${styles.shadowDefault};
                opacity: 0.5;
                background: ${styles.colorMain};
                cursor: not-allowed;
            `}
            
            ${props => props.flat && css`
                -webkit-box-shadow: none;
                -moz-box-shadow: none;
                box-shadow: none;
                background: rgba(0, 0, 0, 0.1);
            `}
            
            ${props => props.flat && props.disabled && css`
                background: none;
            `}
        }
    }
    
    &:focus {
        outline: none;
    }  
`;
