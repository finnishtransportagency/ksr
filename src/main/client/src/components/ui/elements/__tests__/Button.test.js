import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from '../Button';

describe('<Button />', () => {
    it('renders correctly', () => {
        const wrapper = renderer.create(<Button />).toJSON();
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).toHaveStyleRule('color', '#F1F1F1');
        expect(wrapper).toHaveStyleRule('background', '#3DB7E4');
    });

    it('sets correct styles with flat prop', () => {
        const wrapper = renderer.create(<Button flat />).toJSON();
        expect(wrapper).toHaveStyleRule('color', '#3DB7E4');
        expect(wrapper).toHaveStyleRule('background', 'none');
    });

    it('sets correct styles with disabled prop', () => {
        const wrapper = renderer.create(<Button disabled />).toJSON();
        expect(wrapper).toHaveStyleRule('opacity', '0.5');
    });
});
