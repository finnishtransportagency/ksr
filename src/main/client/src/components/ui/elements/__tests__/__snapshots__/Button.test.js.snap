// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
.c0 {
  text-transform: uppercase;
  font-weight: 700;
  border: 0;
  padding: 1rem;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  margin: 1em 0;
  color: #F1F1F1;
  background: #3DB7E4;
  -webkit-box-shadow: 0 2px 4px 0 hsla(0,0%,0%,0.4);
  -moz-box-shadow: 0 2px 4px 0 hsla(0,0%,0%,0.4);
  box-shadow: 0 2px 4px 0 hsla(0,0%,0%,0.4);
}

.c0:hover {
  cursor: pointer;
  -webkit-box-shadow: 0 2px 6px 0 hsla(0,0%,0%,0.6);
  -moz-box-shadow: 0 2px 6px 0 hsla(0,0%,0%,0.6);
  box-shadow: 0 2px 6px 0 hsla(0,0%,0%,0.6);
  background: #00B0CA;
}

.c0:focus {
  outline: none;
}

<button
  className="c0"
/>
`;
